src/storage/mod.rs

```diff
@@ -1073,6 +1199,72 @@ impl Storage {
             .flatten()
     }
 
+    pub fn async_raw_batch_scan(
+        &self,
+        ctx: Context,
+        start_keys: Vec<Vec<u8>>,
+        each_limit: usize,
+        key_only: bool,
+    ) -> impl Future<Item = Vec<Result<KvPair>>, Error = Error> {
+        const CMD: &str = "raw_batch_scan";
+        let engine = self.get_engine();
+        let priority = readpool::Priority::from(ctx.get_priority());
+
+        let start_keys: Vec<Key> = start_keys.into_iter().map(Key::from_encoded).collect();
+
+        let res = self.read_pool.future_execute(priority, move |ctxd| {
+            let mut _timer = {
+                let ctxd = ctxd.clone();
+                let mut thread_ctx = ctxd.current_thread_context_mut();
+                thread_ctx.start_command_duration_timer(CMD, priority, true)
+            };
+
+            Self::async_snapshot(engine, &ctx)
+                .and_then(move |snapshot: Box<Snapshot>| {
+                    let mut thread_ctx = ctxd.current_thread_context_mut();
+                    let _t_process = thread_ctx.start_processing_read_duration_timer(CMD);
+
+                    let mut statistics = Statistics::default();
+                    let mut result = Vec::new();
+                    for start_key in &start_keys {
+                        let pairs = Storage::raw_scan(
```

Comment:
There may be duplicated keys in result. For example, when the underlying keys are `a1`,`a2`,`a3`,`b1`,`c1`,`c2` and `c3`, when you call batch scan with start keys are `a`,`b` and `c`, and each_limit is 3, the result will contains keys `a1`,`a2`,`a3`,`b1`,`c1`,`c2`,`c1``c2` and `c3`.

Reply:
So let's change it to start key and end key pairs and keep each limit. How is that?

Reply:
I have changed it to limit scan with start_key and end_key. Please check it out. And the blank line, I just choose one style used in this file. Hope it is ok.

Reply:
Updated to use one KeyRange field instead of two separate start/end keys fields